src/awkward/_v2/contents/listoffsetarray.py

```diff
@@ -1778,15 +1500,15 @@ def _reduce_next(
                     outstarts.data,
                     outstops.data,
                     distincts.data,
-                    distincts_length,
+                    distincts.length,
                     gaps.data,
                     outlength,
                 )
             )
 
             if reducer.needs_position:
-                nextshifts = ak._v2.index.Index64.empty(nextlen, self._nplike)
-                nummissing = ak._v2.index.Index64.empty(maxcount[0], self._nplike)
+                nextshifts = ak._v2.index.Index64.empty(nextcarry.length, self._nplike)
+                nummissing = ak._v2.index.Index64.empty(maxcount, self._nplike)
```

Comment:
I was wondering about how `maxcount` lost its `[0]`, but now I see that `_rearrange_prepare_next` returns the scalar value, rather than the one-element array.

This might come up again if `maxcount` is ever a TypeTracerArray. But the beauty of this solution is that we'd only have to fix it in one place (in the `_rearrange_prepare_next` function) if we do have to change it.